gst/nnstreamer/elements/gsttensor_debug.c

```diff
@@ -0,0 +1,441 @@
+ */
+static void
+gst_tensor_debug_class_init (GstTensorDebugClass * klass)
+{
+  GObjectClass *object_class;
+  GstElementClass *element_class;
+  GstBaseTransformClass *trans_class;
+
+  GST_DEBUG_CATEGORY_INIT (gst_tensor_debug_debug, "tensor_debug", 0,
+      "Element to provide debug information of other/tensors streams. If this is enabled, the pipeline performance and overhead may be deteriorated significantly.");
+
+  trans_class = (GstBaseTransformClass *) klass;
+  object_class = (GObjectClass *) klass;
+  element_class = (GstElementClass *) klass;
+
+  /* GObjectClass vmethods */
+  object_class->set_property = gst_tensor_debug_set_property;
+  object_class->get_property = gst_tensor_debug_get_property;
+  object_class->finalize = gst_tensor_debug_finalize;
+
+  /**
+   * GstTensorDebug::silent:
+   *
+   * The flag to enable/disable debugging messages.
+   */
+  g_object_class_install_property (object_class, PROP_SILENT,
+      g_param_spec_boolean ("silent", "silent", "Produce verbose output",
+          DEFAULT_SILENT, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  /**
+   * GstTensorDebug::output:
+   *
+   * The combination of enums configuring output methods.
+   * @todo check the behavior of name and nick (output methods vs output)
+   */
+  g_object_class_install_property (object_class, PROP_OUTPUT,
+      g_param_spec_flags ("output-method", "output",
+          "Output methods for debug/profile contents. Different methods can be enabled simultaneously.",
+          TENSOR_DEBUG_TYPE_OUTPUT_FLAGS, DEFAULT_TENSOR_DEBUG_OUTPUT_FLAGS,
+          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  /**
+   * GstTensorDebug::cap:
+   *
+   * The logging preference of the stream capability (GSTCAP).
+   */
+  g_object_class_install_property (object_class, PROP_CAP,
+      g_param_spec_enum ("capability", "cap",
+          "The logging preference for stream capability (GSTCAP)",
+          TENSOR_DEBUG_TYPE_CAPS, DEFAULT_TENSOR_DEBUG_CAP,
+          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+  /**
+   * GstTensorDebug::meta:
+   *
+   * The logging preference of in-stream metadata (GSTMETA).
+   */
+  g_object_class_install_property (object_class, PROP_META,
+      g_param_spec_flags ("metadata", "meta",
+          "The logging preference for stream metadata (GstMeta)",
+          TENSOR_DEBUG_TYPE_META_FLAGS, DEFAULT_TENSOR_DEBUG_META_FLAGS,
+          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
+
+
+  /* set pad template */
+  gst_element_class_add_pad_template (element_class,
+      gst_static_pad_template_get (&src_factory));
+  gst_element_class_add_pad_template (element_class,
+      gst_static_pad_template_get (&sink_factory));
+
+  gst_element_class_set_static_metadata (element_class,
+      "TensorDebug",
+      "Filter/Tensor",
+      "Help debug or profiel a tensor stream by logging the desired details of other/tensors. Users may log the details to console, files, or memory buffers.",
```

Comment:
minor typo - profiel, fix this in next PR.

Reply:
Fixed.